docs/core/compatibility/core-libraries/6.0/partial-byte-reads-in-streams.md

```diff
@@ -1,62 +1,108 @@
+The `Read()` and `ReadAsync()` methods on <xref:System.IO.Compression.DeflateStream>, <xref:System.IO.Compression.GZipStream>, and <xref:System.Security.Cryptography.CryptoStream> might no longer return as many bytes as were requested.
+
+Previously, <xref:System.IO.Compression.DeflateStream>, <xref:System.IO.Compression.GZipStream>, and <xref:System.Security.Cryptography.CryptoStream> diverged from typical <xref:System.IO.Stream.Read%2A?displayProperty=nameWithType> and <xref:System.IO.Stream.ReadAsync%2A?displayProperty=nameWithType> behavior in the following two ways, both of which this change addresses:
 
 - They didn't complete the read operation until either the buffer passed to the read operation was completely filled or the end of the stream was reached.
 - As wrapper streams, they didn't delegate zero-length buffer functionality to the stream they wrap.
 
-This change addresses both of these issues.
+Consider this example that creates and compresses 150 random bytes. It then sends the compressed data one byte at a time from the client to the server, and the server decompresses the data by calling `Read` and requesting all 150 bytes.
+
+:::code language="csharp" source="./snippets/ReadStream.cs":::
+
+In previous versions of .NET and .NET Framework, the following output shows that `Read` was only called once. Even though data was available for `GZipStream` to return, `Read` was forced to wait until the requested number of bytes was available.
+
+```output
+Read: 150 bytes
+Total received: 150 bytes
+```
+
+In .NET 6 and later versions, the following output shows that `Read` was called multiple times until *all* the requested data was received. Even though the call to `Read` requests 150 bytes, each call to `Read` was able to successfully decompress *some* bytes (that is, all of the bytes that had been received at that time) to return, and it did:
+
+```output
+Read: 1 bytes
+Read: 101 bytes
+Read: 4 bytes
+Read: 4 bytes
+Read: 2 bytes
+Read: 2 bytes
+Read: 2 bytes
+Read: 2 bytes
+Read: 3 bytes
+Read: 2 bytes
+Read: 3 bytes
+Read: 2 bytes
+Read: 2 bytes
+Read: 2 bytes
+Read: 2 bytes
+Read: 1 bytes
+Read: 2 bytes
+Read: 1 bytes
+Read: 1 bytes
+Read: 1 bytes
+Read: 2 bytes
+Read: 1 bytes
+Read: 1 bytes
+Read: 2 bytes
+Read: 1 bytes
+Read: 1 bytes
+Read: 2 bytes
+Total received: 150 bytes
+```
 
 ## Old behavior
 
-When `Stream.Read` or `Stream.ReadAsync` was called on one of the affected stream types with a buffer of length `N`, the operation would not complete until:
+When `Stream.Read` or `Stream.ReadAsync` was called on one of the affected stream types with a buffer of length `N`, the operation wouldn't complete until:
 
 - `N` bytes had been read from the stream, or
-- The underlying stream they wrap returned 0 from a call to its read, indicating no more data is available.
+- The underlying stream they wrap returned 0 from a call to its read, indicating no more data was available.
```

Comment:
I think the phrase `underlying stream` indicates it's wrapped, so this is redundant?
```suggestion
- The underlying stream returned 0 from a call to its read, indicating no more data was available.
```